### DISCUSSION TITLE
Conditional fields in Rename Template

### DISCUSSION THREAD

**Discussion by HudsonHawk** [comictagger/comictagger#684] [Q&A] [answered]
I have encountered an issue when attempting to use a custom Template in the _Rename_ tab within _Settings_. My template looks like this:
`{series_group[0]}\{series_group}\{series} #{issue} ({year})`

The idea being that I want to categorize my _series_group_ values within subfolders that are named according to the first letter of the _series_group_.
However, when the _series_group_ has no value, I see the following error message:
![image](https://github.com/user-attachments/assets/7729fa87-33f5-4bd1-bc49-51f309947bba)

My next thought was to try to create a conditional inline string:
`{series_group[0] if series_group else ''}\{series_group}\{series} #{issue} ({year})`
but the application will not allow me to set this value. The following error message is shown below the _Template_ text box:

**Only '.' or '[' may follow ']' in format field specifier**

Is there a way to conditionally set a format specifier in the template field?

**Comment by lordwelch** [comment, marked as answer]
First these aren't [f-strings] they are [format strings]. Currently you can't do that as it errors when series_group is empty and you try to get the first character. The next alpha will handle the error properly and let you do `{series_group[0]+Unknown}\{series_group+Unknown}\{series} #{issue} ({year})`
you can still do `{series_group+Unknown}` right now but that won't let you categorize by the first letter

[f-strings]: https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals
[format strings]: https://docs.python.org/3/library/string.html#format-string-syntax

**Reply by HudsonHawk** [reply]
Thanks for the explanation.

**Reply by lordwelch** [reply]
1.6.0-beta.1 is now released, let me know if you run into any more issues
